src/HDInsight/HDInsight/ManagementCommands/EnableAzureHDInsightAzureMonitorCommand.cs

```diff
@@ -0,0 +1,141 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.WindowsAzure.Commands.Common;
+using Microsoft.Azure.Commands.HDInsight.Commands;
+using Microsoft.Azure.Management.HDInsight.Models;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.HDInsight.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.HDInsight
+{
+    [Cmdlet("Enable", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "HDInsightAzureMonitor", DefaultParameterSetName = SetByNameParameterSet, SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    public class EnableAzureHDInsightAzureMonitorCommand : HDInsightCmdletBase
+    {
+        #region Input Parameter Definitions
+
+        private const string SetByNameParameterSet = "SetByNameParameterSet";
```

Comment:
Please use appropriate parameter set names.

Reply:
> Please use appropriate parameter set names.

Hi @msJinLei, there are many other service also are using this parameter set. Can we keep this?

Reply:
changed to bynameparameterset